client/mysql.cc

```diff
@@ -1924,6 +1924,16 @@ get_one_option(const struct my_option *opt, const char *argument, const char *)
     else
       verbose++;
     break;
+  case 'w':
+    if (argument == disabled_my_option)
+      wait_flag= 0;
+    else
+    {
+      wait_flag= 1;
+      if (argument)
+        wait_retries= atoi(argument);
```

Comment:
that is, no argument means "retry once"? Looks very weird. As far as I can see you have
```
$ mariadb
```
will retry forever, but
```
$ mariadb -w
```
will retry once?

Why not to do it as in `mariadb-admin` ? By default no retry. `-w` means retry forever. `-w number` means retry that many times.

Except that `--skip-wait` should mean "no retry" as you did above, and `mariadb-admin` doesn't have it, I suggest you fix there.